drv/lpc55-rng/src/main.rs

```diff
@@ -197,14 +225,96 @@ impl NotificationHandler for Lpc55RngServer {
     }
 }
 
+/// Load a type implementing HandoffData (and others) from a config::DataRegion.
+/// Errors will be reported in the ringbuf and will return None.
+#[cfg(feature = "dice-seed")]
+fn load_data_from_region<
+    T: for<'a> Deserialize<'a> + HandoffData + SerializedSize,
+>(
+    region: &DataRegion,
+) -> Option<T> {
+    use core::slice;
+
+    // Safety: This memory is setup by code executed before hubris and
+    // exposed using the kernel `extern-regions` mechanism. The safety of
+    // this code is an extension of our trust in the hubris pre-main, kernel,
+    // and build process.
+    let data = unsafe {
+        slice::from_raw_parts(region.address as *mut u8, region.size)
+    };
+
+    // this can be replaced w/ .ok() if we get rid of the ringbuf entry
```

Comment:
**suggestion:** `inspect_err` could be used to do the ringbuf, chained to `ok()`